### PR TITLE
add an option to skip validation of status

### PR DESCRIPTION
**Pull request by alex-laties**
shall we?

Since the status:state can differ from install to install, we add an option to ignore the client side check.

### CODE REVIEW

**Comment by william-richard** [comment]
I think I would prefer if we just asked the installation for the list of valid states, especially since there's already an API endpoint that gives it to you.
http://tumblr.github.io/collins/api.html#api-state-get-all-section

**Comment by roymarantz** [comment]
+1 to getting the list of valid states and using that

**Comment by byxorna** [comment]
+1 on @Primer42's suggestion. We still want to validate statuses (as those are fixed), but states we probably want to do validation by querying collins first, before trying to use the state provided. That way if the user provides a bogus state, we can just dump the list of states we just queried.

**Comment by alex-laties** [comment]
@Primer42 @roymarantz @byxorna added pulling valid states from the remote for validation.

I still believe there's edge-casey value for having the ability to skip client-side validation, so I'm leaving it in for now.

File: lib/collins/cli/modify.rb [hunk 1]
```diff
@@ -32,6 +25,31 @@ def initialize
       @parsed = false
     end
 
+    def valid_states
+      return @_stored_valid_states unless @_stored_valid_states.nil? #don't have to rebuild each time
+      remote_states = collins.state_get_all
```
Comment (byxorna) [inline lib/collins/cli/modify.rb:30]: so, my hesitation here is that just running `collins modify -h` will blow up if you dont have a correctly configured collins.yaml (or if you are missing the network). The help output shouldnt be doing any network access.

**Comment by byxorna** [comment]
maybe we shouldnt be doing any state validation here at all, and let collins API tell us if we fucked up. Perhaps pushing the "show me valid states and statuses" could be rolled into collins find instead?

**Comment by alex-laties** [comment]
@byxorna two different notes:
- I definitely think it makes more sense to just let the API validate than the client.
- The help text blowing up when collins.yaml/network is down is kind of a communication issue (what error gets reported), not a functionality issue. `collins-cli` is completely hosed anyway if either the config or network is messed up. Can definitely have better messaging around why the failure occurred though.

I'm definitely in favor of just stripping the validation for states. It's simpler, cleaner, and we'd have to trust the server anyway for the client-side validation.

**Comment by roymarantz** [comment]
In my ideal world we would add a list-state function to explicitly list the valid states, i.e. do the query.  Also at option parse time it should try to get the valid states and validate if it was able.  I would also like to have completion (bash level) of the command and state using this information.  (I don't ask for much :-) )

**Comment by byxorna** [comment]
For example, running a binary which requires root with `--help` doesnt say "fuck you, you arent root". It just shows the help. All the collins-cli commands should have a functional --help output, regardless of your network status, collins configuration yaml, etc.

**Comment by alex-laties** [comment]
Closing this, since what we really want is to strip the state validation stuff. Maybe add a 'list available states' thing.

**Pull request closed without merging**